contracts/MyMultiSig.cdc

```diff
@@ -57,74 +66,60 @@ pub contract MyMultiSig {
             let keyList = Crypto.KeyList()
             let account = getAccount(acctAddress)
 
-            let publicKeys: [[UInt8]] = []
-            let weights: [UFix64] = []
-            let signAlgos: [UInt] = []
+            
             
             let uniqueKeys: {Int: Bool} = {}
             for id in keyIds {
                 uniqueKeys[id] = true
             }
             assert(uniqueKeys.keys.length == keyIds.length, message: "Invalid duplicates of the same keyID provided for signature")
 
-            var counter = 0
+            // In verify we need a [KeyListSignature] so we do that here
+            let signatureSet: [Crypto.KeyListSignature] = []
             var totalWeight = 0.0
-            while (counter < keyIds.length) {
-                let accountKey: AccountKey = account.keys.get(keyIndex: keyIds[counter]) ?? panic("Provided key signature does not exist")
+
+            var i = 0
+            while (i < keyIds.length) {
```

Comment:
nit: it's easier to iterate like this, so we don't have to use the keyIds[i]
`for key in keyIds`

Reply:
ignore this one, I see that we are using this variable a lot 😅